api/src/generate-service-worker.js

```diff
@@ -0,0 +1,137 @@
+const swPrecache = require('sw-precache');
+const path = require('path');
+const fs = require('fs');
+const crypto = require('crypto');
+
+const environment = require('./environment');
+const db = require('./db');
+const logger = require('./logger');
+
+const SWMETA_DOC_ID = 'service-worker-meta';
+const apiSrcDirectoryPath = __dirname;
+
+const staticDirectoryPath = environment.getExtractedResourcesPath();
+const scriptOutputPath = path.join(staticDirectoryPath, 'js', 'service-worker.js');
+
+/**
+ * Generates a sha1 of the existent service worker file.
+ * If the file does not exist, it returns undefined.
+ * If any error occurs when reading, it returns undefined.
+ * If any error occurs when generating the hash, it returns undefined.
+ * @return {Promise<string|undefined>}
+ */
+const getServiceWorkerHash = () => {
+  return new Promise((resolve) => {
+    try {
+      if (!fs.existsSync(scriptOutputPath)) {
+        return resolve();
+      }
+
+      const hash = crypto.createHash('sha1');
+      const stream = fs.createReadStream(scriptOutputPath);
+      stream.setEncoding('utf8');
+      stream.on('data', (data) => hash.update(data));
+      stream.on('end', () => resolve(hash.digest('hex')));
+      stream.on('error', (err) => {
+        logger.error('Error while reading service worker: %o', err);
+        resolve();
+      });
+    } catch (err) {
+      logger.error('Error while generating service worker hash: %o', err);
+      resolve();
+    }
+  });
+};
+
+const getLoginPageContents = async () => {
+  // avoid circular dependency
+  const loginController = require('./controllers/login');
```

Comment:
Yea, me too. The issue here is the common code in `config.js` (config requires service worker, which requires login controller, which requires config). 

I'll try a refactor and see how it looks like.

Reply:
Ok, I've pulled out all the "watch" logic from `config.js` into a `bootstrap` service. 
Config now just has all it's previous used exports to get settings and translations. 
What do you think?

Reply:
Yes, perfect. I think the naming could be better because we use "bootstrap" everywhere... it's almost as bad as "utils"! But it's difficult to name because it's trying to do too much right now. Something like `db-watcher` may be better?

Reply:
`config-watcher`? `config-listener`? `db-watcher` might be too generic, since this only handles propagating configuration changes to the process.

Reply:
Went with `config-watcher`.